packages/core/src/tools/tool-registry.ts

```diff
@@ -416,7 +416,7 @@ export class ToolRegistry {
         );
       }
     } catch (e) {
-      console.error(`Tool discovery command "${discoveryCmd}" failed:`, e);
+      debugLogger.warn(`Tool discovery command "${discoveryCmd}" failed:`, e);
```

Comment:
this seems like a debugLogger.error

it gets rethrown otherwise would say it needs to be surfaced directly to the user.